docs/rfc/active/5818-serial-updates.md

```diff
@@ -0,0 +1,313 @@
+* Potentially blocking MCE Kafka queue
+
+[![](https://mermaid.ink/img/pako:eNqNksFOwzAMhl8lygm07QV6GEIw7QDTUKuJSy9W4tGINimpc0DT3h136dZQjYkequj3p9-_nRykchplJjv8CmgVPhv48NCUVvAHipwXT7VBS1GJ58VyOVtvikyskR67FhW9G6penfqMFNd6JJZ6ORP9f-dtlFJqgwQaCApuhYljZH5VF4yf2qrgPceYmnE55psQs_qSbMifkG9AqhKQWE1m3LUcAKPTXe90fyN-Ct-aANqWu-iHYc81nZUoXJlo-xJLWHco0Hvn_0RXeb7NB9rqq0PlWCN0ON5QOlN6bwN4ubqRdS0JMg26QGOSEWKfxf-MOKOcywZ9A0bzUzz0cimpwgZLmfFR4x5CTaUs7ZFRCOSKb6tkRj7gXIbT0oeXK7M98IrO6kobXvwgHn8AOC71Xg)](https://mermaid.live/edit#pako:eNqNksFOwzAMhl8lygm07QV6GEIw7QDTUKuJSy9W4tGINimpc0DT3h136dZQjYkequj3p9-_nRykchplJjv8CmgVPhv48NCUVvAHipwXT7VBS1GJ58VyOVtvikyskR67FhW9G6penfqMFNd6JJZ6ORP9f-dtlFJqgwQaCApuhYljZH5VF4yf2qrgPceYmnE55psQs_qSbMifkG9AqhKQWE1m3LUcAKPTXe90fyN-Ct-aANqWu-iHYc81nZUoXJlo-xJLWHco0Hvn_0RXeb7NB9rqq0PlWCN0ON5QOlN6bwN4ubqRdS0JMg26QGOSEWKfxf-MOKOcywZ9A0bzUzz0cimpwgZLmfFR4x5CTaUs7ZFRCOSKb6tkRj7gXIbT0oeXK7M98IrO6kobXvwgHn8AOC71Xg)
+
+### 6: PATCH with JSON-Delta Format
+
+Client sends its update request in JSON-delta or some other delta format. GMS will reject the update if the previous 
+side of the delta does not match.
+
+## How we teach this
+
+> What names and terminology work best for these concepts and why? How is this idea best presented? As a continuation
+> of existing DataHub patterns, or as a wholly new one?
+
+> What audience or audiences would be impacted by this change? Just DataHub backend developers? Frontend developers?
+> Users of the DataHub application itself?
+
+> Would the acceptance of this proposal mean the DataHub guides must be re-organized or altered? Does it change how
+> DataHub is taught to new users at any level?
+
+> How should this feature be introduced and taught to existing audiences?
+
+We would have two potential additions to terminology:
+1. Previous state: The version(s) of aspects required in order for a GM update to succeed.
+2. Preconditions: A wider set of assertions which must be true in order for a GM update to succeed.
+
+We don't really cover "Preconditions" in this RFC, and in fact I argue we should never do, as this leaks business 
+knowledge into the GMS code. I would therefore use "Previous State" to describe the aspect versions required for 
+updates.
+
+### Documentation
+
+This should be added as an extra section of the following:
+* REST API examples
+* Java/Python REST Emitter
+* GraphQL mutations?
+
+## Drawbacks
+
+> Why should we *not* do this? Please consider the impact on teaching DataHub, on the integration of this feature with
+> other existing and planned features, on the impact of the API churn on existing apps, etc.
+
+> There are tradeoffs to choosing any path, please attempt to identify them here.
+
+There are two major reasons this will be difficult to achieve: Aspect Versioning and Performance.
+
+### Aspect Versioning
+
+Currently, [aspect versioning](/docs/advanced/aspect-versioning.md) uses 0 as the latest aspect version. We would 
```

Comment:
We can just use total aspect count, instead of version. Serves the same purpose

Reply:
As discussed on video chat, this might not work if we limit the retention of older versions. We need something that is deterministic and not subject to race conditions.